database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -75,39 +78,39 @@ private int getIndexForKey(String key) {
     }
 
     private boolean isKeyAtIndex(String key, int index) {
-        return index >= 0 && index < getCount() && mDataSnapshots.get(index).getKey().equals(key);
+        return index >= 0 && index < size() && mDataSnapshots.get(index).getKey().equals(key);
     }
 
     @Override
     public void onChildAdded(DataSnapshot keySnapshot, String previousChildKey) {
-        super.setOnChangedListener(null);
+        setShouldNotifyListeners(false);
```

Comment:
I don't have the solution, but I always felt that the "remove listener, do operation, add listener" pattern was code smell.  Can we do anything cleaner while we're already in here making all these API changes?  `shouldNotifyListeners` is not much of an improvement (although it's definitely something).